sdk/core/azure-core/inc/azure/core/url.hpp

```diff
@@ -239,5 +239,55 @@ namespace Azure { namespace Core {
      * @return Absolute URL with URL-encoded query parameters.
      */
     std::string GetAbsoluteUrl() const;
+
+    /**
+     * @brief Compares with another instance of Url for equality.
+     * @param other another instance of Url.
+     * @return `true` if values of two Url instances are equal, `false` otherwise.
+     *
+     */
+    bool operator==(Url const& other) const
+    {
+      if (this == &other)
+      {
+        return true;
+      }
+
+      if (m_port != other.m_port
+          || m_encodedQueryParameters.size() != other.m_encodedQueryParameters.size()
+          || m_scheme != other.m_scheme || m_host != other.m_host)
+      {
+        return false;
+      }
+
+      size_t const lhsPathLength = m_encodedPath.length();
+      size_t const rhsPathLength = other.m_encodedPath.length();
+
+      size_t const lhsPathStart = (lhsPathLength > 0 && m_encodedPath[0] == '/') ? 1 : 0;
+      size_t const rhsPathStart = (rhsPathLength > 0 && other.m_encodedPath[0] == '/') ? 1 : 0;
+
+      if ((lhsPathLength - lhsPathStart) != (rhsPathLength - rhsPathStart))
+      {
+        return false;
+      }
+
+      for (size_t l = lhsPathStart, r = rhsPathStart; l < lhsPathLength; ++l, ++r)
+      {
+        if (m_encodedPath[l] != other.m_encodedPath[r])
+        {
+          return false;
+        }
+      }
+
+      return m_encodedQueryParameters == other.m_encodedQueryParameters;
```

Comment:
A little bit cleaner if this check was part of the port, encodedeQuery, scheme, ... checks. Then if you get to the end you return true;

Unless there is a perf win to not early compare the encodedQueryParameters

Reply:
Yes, that is exactly the reason, you can see we're comparing from the least expensive to more and more expensive, and comparing two dictionaries is the most expensive operation, so it is the last.